docs/specs/single-funded-channel.md

```diff
@@ -0,0 +1,115 @@
+# Single-Funded Channel
+
+This feature allows the party who sends the `OpenChannelV2` message (the **initiator**) to provide all the required assets for opening a channel. The counterparty (the **non-initiator**) does not need to contribute funds during channel creation, but must refund the negotiated amount of assets to generate their settlement output cell when the channel is closed.
+
+## Feature Negotiation
+
+- Nodes that support single-funded channels MAY advertise the optional feature bit `ACCEPT_SINGLE_FUNDED_CHANNEL(5)`.
+- Nodes that require peers to accept single-funded channels MUST advertise the mandatory feature bit `ACCEPT_SINGLE_FUNDED_CHANNEL(4)`.
+
+The initiator must not send `OpenChannelV2` to peers that have not advertised either feature bit. The initiator may use OpenChannelV2 with peers that have advertised bit 4 or 5 to open either a single-funded or a dual-funded channel.
+
+## `OpenChannelV2` Schema
+
+```
+table OpenChannelV2 {
+    v1: OpenChannel,
+    refund_reserved_ckb_amount: Uint64,
+}
+
+union FiberMessage {
+    ...
+    AnnouncementSignatures,
+    // Append OpenChannelV2 after AnnouncementSignatures
+    OpenChannelV2,
+}
+```
+
+- `v1` embeds all fields from `OpenChannel`.
+- `refund_reserved_ckb_amount` specifies how many shannons (or SUDT/XUDT units) the non-initiator must refund at channel closure:
+    - Set to zero for dual-funded channels.
+    - Set to non-zero to open single-funded CKB or SUDT/XUDT channels.
+
+The initiator's nested field `v1.reserved_ckb_amount` must cover both the settlement outputs and fees. Given a commitment fee rate $F$, and maximum transaction size $N$, the initiator must reserve a fee of $3FN$.
+
+## Non-Initiator Validation
+
+Upon receiving `OpenChannelV2`, the non-initiator must reject the request if:
+
+1. It has not enabled feature bit 4 or 5.
+2. The ratio `refund_reserved_ckb_amount / minimum_reserved_ckb_amount` falls outside acceptable bounds. Peers configure acceptable ratios in their config files.
+
+Here, `minimum_reserved_ckb_amount = 8 + l + t + d` where:
+
+- 8 for capacity field (shannons)
+- l = lock field size = max(non-initiator shutdown script size, 81)
+- t = type field size = 0 for CKB channels or the SUDT/XUDT script size
+- d = data field size = 0 for CKB or 8 for SUDT/XUDT
+
+## Initiator Validation
+
+Upon receiving `AcceptChannel`, the initiator must abandon the channel if:
+
+1. The `reserved_ckb_amount` is not zero.
+
+## Settlement Transaction Construction
+
+Both dual- and single-funded channels use a similar settlement structure. Each party’s output cell contains:
+
+- **Balance** $B_1$ for the initiator, $B_2$ for the non-initiator.
+- **Reserved amount** $R_1$ for the initiator, $R_2$ for the non-initiator (excluding transaction fees)
+
+For single-funded channels, $R_2 = 0$. Let:
+
+- $A$ = `refund_reserved_ckb_amount`
+- $C$ = `minimum_reserved_ckb_amount`
+
+### Case: $B_2 - A \ge 0$
```

Comment:
The two construction is based on `B2 - A`, however we can simplify the condition by only check `A == 0`.

1. If `A == 0`, both sides use shutdown script.
2. If `A != 0`, non-initiator use lease script.